File: app/src/main/java/com/example/smarthome/ligths/services/ILightRetrofitService.java
package com.example.smarthome.ligths.services;

import com.example.smarthome.ligths.models.Light;
import com.google.gson.JsonObject;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ILightRetrofitService {

    @POST("/lights")
    Call<JsonObject> addLight();

    @DELETE("/lights/{id}")
    Call<JsonObject> deleteLightById(@Path("id") Long id);

    @GET("/lights")
    Call<JsonObject> getLights();

    @GET("/lights/{id}")
    Call<JsonObject> getLightById(@Path("id") Long id);

    @GET("/lights/ip/{ip}")
    Call<JsonObject> getLightByIp(@Path("ip") String ip);

    @PUT("lights/{id}")
    Call<JsonObject> updateLightById(@Path("id") Long id, Light light);

}

File: app/src/main/java/com/example/smarthome/ligths/services/ILightRetrofitService.kt
package com.example.smarthome.ligths.services

import com.example.smarthome.ligths.models.Light
import com.google.gson.JsonObject
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

//TODO update CALL return type
interface ILightRetrofitService {

    @GET("/lights")
    fun getAll(): Call<List<Light>>

    @GET("/lights/{id}")
    fun getById(@Path("id") id: Long) : Call<Light>

    @GET("/lights/ip/{ip}")
    fun getByIp(@Path("ip") ip: String) : Call<Light>

    @POST("/lights")
    fun add(@Body light: Light) : Call<JsonObject>

    @POST("/lights/{id}/turn")
    fun turn(@Path("id") id: Long) : Call<ResponseBody>

    @PUT("/lights/{id}")
    fun updateById(@Path("id") id: Long, @Body light: Light) : Call<ResponseBody>

    @DELETE("/lights/{id}")
    fun deleteById(@Path("id") id: Long) : Call<ResponseBody>

}